JSONObject/JSONObject.swift

```diff
@@ -78,39 +32,74 @@ public enum JSONObject {
-		case is NSNull:
-			return .null
-		case is Int:
-			return .number(object as! JSONNumber)
-		case is UInt:
-			return .number(object as! JSONNumber)
-		case is Float:
-			return .number(object as! JSONNumber)
-		case is Double:
-			return .number(object as! JSONNumber)
-		case is Bool:
-			return .bool(object as! Bool)
-		case is String:
-			return .string(object as! String)
-		case is [Any]:
-			return .array((object as! [Any]).map(JSONObject.with))
-		case is [String:Any]:
-			return .dict((object as! [String:Any])
-				.map { ($0,JSONObject.with($1)) }
-				.reduce([:]) {
-					var m_accumulation = $0
-					m_accumulation[$1.0] = $1.1
-					return m_accumulation
-			})
-		default:
-			return .null
-		}
-	}
+    public static func with(_ object: Any) -> JSONObject {
+        
+        switch object {
+        case is NSNull:
+            return .null
+        case is Int:
+            return .number(object as! JSONNumber)
+        case is UInt:
+            return .number(object as! JSONNumber)
+        case is Float:
+            return .number(object as! JSONNumber)
+        case is Double:
+            return .number(object as! JSONNumber)
+        case is Bool:
+            return .bool(object as! Bool)
+        case is String:
+            return .string(object as! String)
+        case is [Any]:
+            return .array((object as! [Any]).map(JSONObject.with))
+        case is [String:Any]:
+            return .dict((object as! [String:Any])
+                .map { ($0,JSONObject.with($1)) }
+                .reduce([:]) {
+                    var m_accumulation = $0
+                    m_accumulation[$1.0] = $1.1
+                    return m_accumulation
+            })
+        default:
+            return .null
+        }
+	}
+    
+    public static func aFunc(_ object: Any) -> JSONResult<JSONObject> {
+        switch object {
+        case is NSNull:
+            return .success(.null)
+        case is Int:
+            return .success(.number(object as! JSONNumber))
+        case is UInt:
+            return .success(.number(object as! JSONNumber))
+        case is Float:
+            return .success(.number(object as! JSONNumber))
+        case is Double:
+            return .success(.number(object as! JSONNumber))
+        case is Bool:
+            return .success(.bool(object as! Bool))
+        case is String:
+            return .success(.string(object as! String))
+        case is [Any]:
+            return (object as! [Any])
+                .traverse(JSONObject.aFunc)
+                .map(JSONObject.array)
+        case is [String:Any]:
+            return (object as! [String:Any])
+                .mapValues(JSONObject.aFunc)
+                .map { ($0,$1) }
+                .traverse { (tuple) -> JSONResult<(String,JSONObject)> in
+                    let (key,result) = tuple
+                    return result.map{ (key,$0) } }
+                .map { JSONObject.dict($0
+                    .reduce([:]) {
```

Comment:
You probably don't need to do this: `Dictionary` has already an initializer from a `Sequence` of key-value pairs, that is, `.init(uniqueKeysWithValues:)`.